src/data/datetime/readable_datetime.R

```diff
@@ -99,35 +99,62 @@ filter_wanted_dates <- function(output, participant_file, device_type){
 
   return(output)
 }
+
+process_by_chunks <- function(timezone_parameters, device_type, pid, participant_file, time_segments, time_segments_type, include_past_periodic_segments, output_file_path){
+  
+  function(x, pos){
+
+    output_is_empty <<- FALSE
+
+    if(timezone_parameters$TYPE == "SINGLE"){
+      output <- x %>% mutate(local_timezone = timezone_parameters$SINGLE$TZCODE)
+      most_common_tz <- timezone_parameters$SINGLE$TZCODE
+    }
+    else if(timezone_parameters$TYPE == "MULTIPLE"){
+      output <- multiple_time_zone_assignment(x, timezone_parameters, device_type, pid, participant_file)
+      most_common_tz <- get_participant_most_common_tz(timezone_parameters$MULTIPLE$TZCODES_FILE, participant_file) # in assign_to_multiple_timezones.R
+    }
+
+    output %<>%  
+      create_mising_temporal_column(device_type)  %>% 
+      split_local_date_time() %>% 
+      assign_to_time_segment(time_segments, time_segments_type, include_past_periodic_segments, most_common_tz) %>% 
+      filter_wanted_dates(participant_file, device_type) %>% 
+      arrange(timestamp)
+    
+    output %>% write_csv(output_file_path, append=ifelse(pos > 1, T, F))
+  }
+}
+
 readable_datetime <- function(){
-  input <- read.csv(snakemake@input[["sensor_input"]]) %>% arrange(timestamp)
+
   time_segments <- read.csv(snakemake@input[["time_segments"]])
   participant_file <- snakemake@input[["pid_file"]]
   device_type <- snakemake@params[["device_type"]]
   timezone_parameters <- snakemake@params[["timezone_parameters"]]
   pid <- snakemake@params[["pid"]]
   time_segments_type <- snakemake@params[["time_segments_type"]]
   include_past_periodic_segments <- snakemake@params[["include_past_periodic_segments"]]
+  output_file_path <- snakemake@output[[1]]
 
   validate_user_timezones(timezone_parameters)
+
+  output_is_empty <<- TRUE
```

Comment:
Is there a way to not depend on a global var? Maybe read_csv_chunked has a callback when the input is empty?

Reply:
Yes! Thanks for your suggestions. I just found that `AccumulateCallback` might be more suitable for us. It accumulates a single result. A parameter named `acc` can be used to specify the initial value of the accumulator. Here, I set acc=0. If the input data is empty, it will return 0; otherwise, it will be >0.